controllers/inverse_kinematics_pyikfast/inverse_kinematics_pyikfast.py

```diff
@@ -80,27 +80,74 @@
+    """Move robot endpoint to a known position that aligns it perfectly with the target"""
+    for i in range(len(joint_state)):
+        motors[i].setPosition(joint_state[i])
+# plase end-effector on target
+move_to_joint_state(motors, target_position_joint_state)
 
+# --------------------------------------------------------------------
+# show that the endpoint pose is indeed the targtet pose
 # Initialize our inverse kinematics module
 ik = inverseKinematics(pyikfast_module, protoName, minPositions, maxPositions)
+print("ik", ik)
+
+fk_pos, fk_rot = ik.get_fk(target_position_joint_state)
+print("fk_pos", fk_pos)
+print("fk_rot", fk_rot)
+
+# check that the target position and rotation matches te end-effector's
+target = supervisor.getFromDef("TARGET")
+# target position relative to world.
+target_pos_world = np.array(target.getPosition())
+# target rotation relative to world
+target_rot_world = np.array(target.getOrientation()).reshape(3, 3)
+assert np.allclose(fk_pos, target_pos_world)
+assert np.allclose(fk_rot, target_rot_world)
+print("Yay, target pose and end-effector pose match")
+
+# --------------------------------------------------------------------
+# Try is get an IK solution for the current end-point pose.
+# there should be at least one solution, target_position_joint_state
+ikResults = ik.get_ik(fk_pos, fk_rot)
+
+assert len(ikResults) > 0
+
+#############################################################################################
+# We don't get any further
+print("Yay, We found in IK solution to our current end effector pose")
+
 
 # Initialize the RelativePositions module. 'TARGET' is the DEF of the spawned sphere.
 # You can change this DEF to any other object. You can also initialilze several like this:
 # RelPos_1 = RelativePositions(supervisor, 'TARGET1')
 # RelPos_2 = RelativePositions(supervisor, 'TARGET2')
-RelPos = RelativePositions(supervisor, 'TARGET')
-
+RelPos = RelativePositions(supervisor, "TARGET")
+print("RelPos.get_pos()", RelPos.get_pos())
 # ---------------------- Main Loop Velocity Control-------------------------
-print('-------------------------------------------------------')
-print('Using Velocity Control to move the end-effector in a circle for 10s.')
+print("-------------------------------------------------------")
+print("Using Velocity Control to move the end-effector in a circle for 10s.")
+
 # Calculating an initial position and orientation defines the starting position of the
 # velocity control inverse kinematic calculations.
-init_pos = np.array([1, 0, 0.5])
-init_rot = np.array([[1.0, 0.0, 0.0], [0.0, -1.0, 0.0], [0.0, 0.0, -1.0]])
-ikResults = ik.get_ik(init_pos, init_rot)
+
+robot_base = supervisor.getSelf()
+robot_rot_base = np.transpose(np.array(robot_base.getOrientation()).reshape(3, 3))
+robot_pos_base = np.array(robot_base.getPosition())
+print("robot_pos_base", robot_pos_base)
+print("robot_rot_base", robot_rot_base)
+
+print("ikResults", ikResults)
```

Comment:
This portion is should be only in your troubleshoot implementation. But there is an argument to be made to figure out a generic initial pose for all sample robots. What works for the rather large irb might not work for the UR10